packages/filesystem/src/browser/filesystem-frontend-contribution.ts

```diff
@@ -30,6 +30,7 @@ import { TreeWidgetSelection } from '@theia/core/lib/browser/tree/tree-widget-se
 import { FileSystemPreferences } from './filesystem-preferences';
 import { FileSelection } from './file-selection';
 import { FileUploadService } from './file-upload-service';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
```

Comment:
filesystem is not responsible to know anything about workspaces, only to provide generic access to fs

functionality dependent on opened workspace should go to workspace extension

BTW how one will upload in the case with multi-root workspace? Should it select to which workspace root to upload first?

Reply:
in case of multi-root workspace, my original idea is to have upload default to the root of the first workspace, which is quite intuitive. But as others point out, a more official way would be using quick-open pop-up to let user select a workspace